### PR TITLE
Set devices to 1 when it's just `Trainer(accelerator='auto')`

### PR DESCRIPTION
**Pull request by kaushikb11**
## What does this PR do?

Set devices to 1 when it's just `Trainer(accelerator='auto')`

## Before submitting

- [ ] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [ ] Did you make sure to **update the documentation** with your changes? (if necessary)
- [x] Did you write any **new necessary tests**? (not for typos and docs)
- [x] Did you verify new and **existing tests pass** locally with your changes?
- [ ] Did you list all the **breaking changes** introduced by this pull request?
- [ ] Did you **update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)**? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

**Comment by kaushikb11** [comment]
I am putting this PR on hold. IMO, if a user just does Trainer(accelerator="auto"), we should crash it and ask them to pass devices=x or gpus if it had GPUs.(like we do right now)

Rather than assigning only one device for them.
For this particular reason, we support `devices="auto"`

@PyTorchLightning/core-contributors

**Comment by carmocca** [comment]
It could also default to `devices="auto"` if only `accelerator="auto"` was passed.

**Review by SkafteNicki — APPROVED** [review]

**Review by Borda — APPROVED** [review]

**Comment by ananthsub** [comment]
Is there any discussion of this feature in an issue or RFC? It is very hard to review this without the broader context of what's trying to be achieved here

**Comment by akihironitta** [comment]
I'm not aware of the context outside this PR at all, but I don't think we should accept this change. If it were more common to use only one device out of multiple devices than using all of them, it'd be fine, but it's not common, right?

In my opinion, it should default to `devices="auto"` as carmocca suggests because I think that's what users (at least I) expect, or at least raise an exception asking users to specify `devices` as kaushikb11 suggests.

**Comment by tchaton** [comment]
IMO, I believe accelerator="auto" should be the default and devices become a required argument.

**Comment by kaushikb11** [comment]
`devices` is set to `auto` when it's not passed. Closing the PR.
